File: src/main/java/com/zunza/buythedip/auth/oauth2/dto/NaverOAuth2Response.java
package com.zunza.buythedip.auth.oauth2.dto;

import java.util.Map;

import com.zunza.buythedip.user.constant.OAuth2Provider;

import lombok.RequiredArgsConstructor;

public class NaverOAuth2Response implements OAuth2Response {
	private final Map<String, Object> response;

	public NaverOAuth2Response(
		Map<String, Object> attribute
	) {
		this.response = (Map<String, Object>) attribute.get("response");
	}

	@Override
	public OAuth2Provider getProvider() {
		return OAuth2Provider.NAVER;
	}

	@Override
	public String getProviderId() {
		return response.get("id").toString();
	}

	@Override
	public String getEmail() {
		return response.get("email").toString();
	}
}

File: src/main/kotlin/com/zunza/buythedip/auth/oauth2/dto/NaverOAuth2Response.kt
package com.zunza.buythedip.auth.oauth2.dto

import com.zunza.buythedip.user.constant.OAuth2Provider

class NaverOAuth2Response(
    private val attribute: Map<String, Any>
) : OAuth2Response {

    private val response = attribute["response"] as Map<String, Any>

    override fun getProvider(): OAuth2Provider {
        return OAuth2Provider.NAVER
    }

    override fun getProviderId(): String {
        return response["id"].toString()
    }

    override fun getEmail(): String {
        return response["email"].toString()
    }
}
